dashboard/test/models/contact_rollups_processed_test.rb

```diff
@@ -54,24 +54,27 @@ class ContactRollupsProcessedTest < ActiveSupport::TestCase
     create :contact_rollups_raw
 
     # Each extraction function will be called once per unique email address
-    ContactRollupsProcessed.expects(:extract_opt_in).once
+    ContactRollupsProcessed.expects(:extract_field).once
     ContactRollupsProcessed.expects(:extract_updated_at).once
 
     ContactRollupsProcessed.import_from_raw_table
   end
 
-  test 'extract_opt_in' do
+  test 'extract_field' do
+    table_name = 'dashboard.email_preferences'
+    field_name = 'opt_in'
+
     test_cases = [
-      {input: {}, expected_output: nil},
-      {input: {'dashboard.email_preferences' => {}}, expected_output: nil},
-      {input: {'dashboard.another_table' => {opt_in: 1}}, expected_output: nil},
-      {input: {'dashboard.email_preferences' => {'opt_in' => 0}}, expected_output: {opt_in: 0}},
-      {input: {'dashboard.email_preferences' => {'opt_in' => 1}}, expected_output: {opt_in: 1}},
-      {input: {'dashboard.email_preferences' => {'opt_in' => nil}}, expected_output: {opt_in: nil}}
+      {input: [{}, nil, nil], expected_output: nil},
+      {input: [{'dashboard.email_preferences' => {}}, table_name, field_name], expected_output: nil},
```

Comment:
Replace all references to `dashboard.email_preferences` by `table_name`, and replace `opt_in` by `field_name`. You can even omit the `_name` suffix.